ls1intum/Artemis
# `Development`: Remove unnecessary server API URL constant from client

### Checklist
#### General



#### Client
none of the checks here apply

### Motivation and Context


The `SERVER_API_URL` is a constant used heavily in the client. However, in practice it's always empty (tested on production, ts1, locally, by looking at code).
Also the fact that it is a declared global variable injected by Webpack makes it difficult to switch to a different code bundling process, which is [on the way](https://github.com/ls1intum/Artemis/pull/6538).

Keeping this variable adds unnecessary complexity to the client code. Therefore, it should be removed.

### Description

This PR removes the `SERVER_API_URL` in all places in the client code.

### Steps for Testing

Click around in the application. I've replaced one variable in the whole app, so if it works in one place, it should work in all others.

We should pay a bit more attention in the following places:
- [Group Chat](https://github.com/ls1intum/Artemis/pull/5822)
- [Exam Mode Live Statistics](https://github.com/ls1intum/Artemis/pull/5343)
- [LTI 1.3 Support](https://github.com/ls1intum/Artemis/pull/5693)

The reason being is that code from those features assumed no trailing slash for `SERVER_API_URL`, which differed from all other places. I'm confident that this won't affect anything, but it's better to double-check.

### Code Review Info
List of files where more than literally just removing the variable usage happens:
- `declarations.d.ts`
- `environment.js`
- `webpack.custom.js`
- `interceptor.util.ts`
- `code-editor-domain-dependent-endpoint.service.ts`
- `build-log.service.ts`
- removed slash
  - `lti13-dynamic-registration.ts`
  - `lti13-exercise-launch.component.ts`
  - `exam-monitoring.service.ts`
  - `group-chat.service.ts`

### Review Progress






#### Code Review
#### Manual Tests
